chainer/training/extensions/evaluator.py

```diff
@@ -0,0 +1,168 @@
+import copy
+
+import six
+
+from chainer.dataset import convert
+from chainer.dataset import iterator as iterator_module
+from chainer import link
+from chainer import reporter as reporter_module
+from chainer.training import extension
+from chainer import variable
+
+
+class Evaluator(extension.Extension):
+
+    """Trainer extension to evaluate models on a validation set.
+
+    This extension evaluates the current models by a given evaluation function.
+    It creates a :class:`~chainer.Reporter` object to store values observed in
+    the evaluation function on each iteration. The report for all iterations
+    are aggregated to :class:`~chainer.DictSummary`. The collected mean values
```

Comment:
It might be better to provide an option to specify the aggregation routine. For the current implementation, it is better to report more primitive values, namely precision and recall, let the evaluator aggregate them, and compute the overall F1 score in another extension to edit the observation dictionary.